subprojects/errorprone/src/main/java/org/mockito/errorprone/bugpatterns/MockitoMockedStatic.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (c) 2017 Mockito contributors
+ * This program is made available under the terms of the MIT License.
+ */
+package org.mockito.errorprone.bugpatterns;
+
+import com.google.auto.service.AutoService;
+import com.google.errorprone.BugPattern;
+import com.google.errorprone.BugPattern.SeverityLevel;
+import com.google.errorprone.VisitorState;
+import com.google.errorprone.bugpatterns.BugChecker;
+import com.google.errorprone.bugpatterns.BugChecker.VariableTreeMatcher;
+import com.google.errorprone.matchers.Description;
+import com.sun.source.tree.VariableTree;
+
+@AutoService(BugChecker.class)
+@BugPattern(
+        name = "MockitoMockedStatic",
+        summary = "Fields or parameters annotated with @Mock in MockedStatic can lead to compilation issues.",
+        explanation = "Fields or parameters annotated with @Mock in MockedStatic can lead to compilation issues.",
+        severity = SeverityLevel.ERROR)
+public class MockitoMockedStatic extends BugChecker implements VariableTreeMatcher {
+
+  @Override
+  public Description matchVariable(VariableTree tree, VisitorState state) {
+    if (tree.getModifiers().getAnnotations().stream()
+            .anyMatch(annotation ->
+                    state.getSourceForNode(annotation).equals("@org.mockito.Mock")
+                            && state.getSourceForNode(tree.getType()).equals("org.mockito.MockedStatic"))) {
```

Comment:
Should this check for MockedConstruction too? Both have the same issue I think in that mocking them results in a lifecycle-bound mock that has to be closed to prevent interfering with external components

Reply:
Would you reckon that should be a separate file or in the same one?

Reply:
Not entirely sure. Might be best to wait for confirmation from one of the maintainers! I'd assume it is almost identical logic minus the naming though so whether it is something that is reusable by just passing the class name and error message or not..?

They might not even want to cover the construction case though, so I'd definitely wait for confirmation :-)

Reply:
Let's check for both in this same check. In that case, let's name the checker `MockitoIncompatibleMockTypes`